core/controllers/base_test.py

```diff
@@ -1423,3 +1469,407 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
+            self, handler, handler_class_name):
+        """This test check whether the default_value provided in schema
+        conforms with the schema.
+
+        Args:
+            handler: BaseHandler. A callable to handle the route.
+            handler_class_name: str. Name of the handler class.
+        """
+        if handler_class_name in self.non_schema_handlers:
+            return
+        if handler.HANDLER_ARGS_SCHEMAS is None:
+            return
+
+        schemas = handler.HANDLER_ARGS_SCHEMAS
+        for request_method, request_method_schema in schemas.items():
+            for arg, schema in request_method_schema.items():
+                if 'default_value' not in schema:
+                    continue
+                default_value = {arg: schema['default_value']}
+                default_value_schema = {arg: schema}
+
+                errors = payload_validator.validate(
+                    default_value, default_value_schema, True)
+                if len(errors) == 0:
+                    continue
+
+                self.log_line(
+                    'Handler: %s, argument: %s, default_value '
+                        'validation failed.' % (handler_class_name, key))
+                if (handler_class_name not in
+                        self.handlers_with_non_conforming_default_schema):
+                    self.handlers_with_non_conforming_default_schema.append(
+                        handler_class_name)
+
+    def _handlers_with_schema_are_not_in_the_schema_requiring_list(
+            self, handler, handler_class_name):
+        """This test checks if a handler contains schema then, handler class
+        name should not be present in SCHEMA_REQUIRING_HANDLERS list.
+
+        Args:
+            handler: BaseHandler. A callable to handle the route.
+            handler_class_name: str. Name of the handler class.
+        """
+
+        if handler_class_name in self.non_schema_requiring_handlers:
+            return
+
+        schema_written_for_request_methods = (
+            handler.HANDLER_ARGS_SCHEMAS !=
+                base.BaseHandler.HANDLER_ARGS_SCHEMAS)
+        schema_written_for_url_path_args = (
+            handler.URL_PATH_ARGS_SCHEMAS !=
+                base.BaseHandler.URL_PATH_ARGS_SCHEMAS)
+        handler_have_schema = (schema_written_for_request_methods and
+            schema_written_for_url_path_args)
+        handler_class_needs_removal = (
+            handler_have_schema and
+                handler_class_name in self.non_schema_handlers)
+
+        if handler_class_needs_removal:
+            self.handlers_to_remove.append(handler_class_name)
+
+    def test_general_functionality_of_schema_validation(self):
+        """This method calls all the test functions written below, in
+        a single place for more readability.
+        """
+        for route in main.URLS:
+            # URLS = MAPREDUCE_HANDLERS + other handlers. MAPREDUCE_HANDLERS
```

Comment:
Please add a TODO/issue to drop this check once the MR handlers are removed (which they will be shortly -- /cc @brianrodri). The reason for this is that maybe we might introduce handlers in the future which are tuples.

Reply:
Done